src/shared/features/AppletSettings/ExportDataSetting/Popups/ExportSettingsPopup/ExportSettingsPopup.tsx

```diff
@@ -26,22 +23,15 @@ export const ExportSettingsPopup = ({ isOpen, onClose, onExport }: ExportSetting
   const { t } = useTranslation('app');
   const { result: appletData } = applet.useAppletData() ?? {};
 
-  const minDate = new Date(appletData?.createdAt ?? '');
-  const getMaxDate = () => getNormalizedTimezoneDate(new Date().toString());
-  const methods = useForm<ExportDataFormValues>({
-    resolver: yupResolver(exportDataSettingSchema() as ObjectSchema<ExportDataFormValues>),
-    defaultValues: {
-      dateType: ExportDateType.AllTime,
-      fromDate: minDate,
-      toDate: getMaxDate(),
-    },
-    mode: 'onSubmit',
-  });
-  const { control, setValue, watch } = methods;
+  const { control, setValue, watch } = useFormContext<ExportDataFormValues>() ?? {};
   const dateType = watch('dateType');
   const fromDate = watch('fromDate');
   const toDate = watch('toDate');
   const hasCustomDate = dateType === ExportDateType.ChooseDates;
+
+  const minDate = new Date(appletData?.createdAt ?? '');
```

Comment:
Could you please pass `minDate` and `getMaxDate` as props from `ExportDataSetting` to avoid code duplication?